Uplift/Core/UpliftApp.swift

```diff
@@ -16,27 +17,56 @@ struct UpliftApp: App {
 
     @UIApplicationDelegateAdaptor(AppDelegate.self) var delegate
     @StateObject var locationManager = LocationManager.shared
+    @StateObject private var mainViewModel = MainView.ViewModel()
 
     // MARK: - UI
 
     var body: some Scene {
         WindowGroup {
-            MainView()
-                .environmentObject(locationManager)
-                .onAppear {
-                    locationManager.requestLocation()
+            ZStack {
+                if mainViewModel.displayMainView {
+                    MainView()
+                        .environmentObject(locationManager)
+                        .onAppear {
+                            locationManager.requestLocation()
+                        }
+                } else if mainViewModel.userDidLogin {
+                    CreateProfileView()
+                        .environmentObject(mainViewModel)
+                } else {
+                    SignInView()
+                        .environmentObject(mainViewModel)
+                        .onAppear {
+                            GIDSignIn.sharedInstance.restorePreviousSignIn { _, _ in
+                            }
+                        }
+                        .onOpenURL { url in
+                            GIDSignIn.sharedInstance.handle(url)
+                        }
                 }
+            }
         }
     }
 
-}
+    class AppDelegate: NSObject, UIApplicationDelegate {
+        func application(
+            _ application: UIApplication,
+            didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+        ) -> Bool {
+            GIDSignIn.sharedInstance.restorePreviousSignIn { user, error in
+                if error != nil || user == nil {
+                    // Show the app's signed-out state.
```

Comment:
Is this section still needed? If so, if you still need to work on this later, use a //TODO: Comment instead